### PR TITLE
Add Type checks

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- fixed types to use python >3.5 style
- fixed type errors
- added stage to travis to check types using pytype

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: rasa_sdk/knowledge_base/storage.py [hunk 1]
```diff
@@ -21,15 +22,14 @@ def __init__(self):
             "8": lambda l: l[7],
             "9": lambda l: l[8],
             "10": lambda l: l[9],
-            "ANY": lambda l: random.choice(list),
+            "ANY": lambda l: random.choice(l),
```
Comment (tmbo) [inline rasa_sdk/knowledge_base/storage.py]: this seemed to be a bug

File: rasa_sdk/executor.py [hunk 2]
```diff
@@ -175,6 +163,8 @@ def _import_submodules(self, package, recursive=True):
                 self._import_submodules(full_name)
 
     def register_package(self, package):
```
Comment (ricwo) [inline rasa_sdk/executor.py]: types here, I guess we can add them in future though

Comment (tmbo) [reply rasa_sdk/executor.py]: added a couple more :+1:

File: rasa_sdk/executor.py [hunk 3]
```diff
@@ -121,11 +107,13 @@ def utter_image_url(self, image, **kwargs):
...
 
-class ActionExecutor(object):
+class ActionExecutor:
     def __init__(self):
         self.actions = {}
 
     def register_action(self, action):
+        from rasa_sdk.interfaces import Action
+
         if inspect.isclass(action):
```
[the remark this answers is not in the document]
Comment (tmbo) [reply rasa_sdk/executor.py]: added them, not sure what the type of a package is though, so used `Any` for now

[the remark this answers is not in the document]
Comment (ricwo) [reply rasa_sdk/executor.py]: should be `types.ModuleType` (thought I commented that further down but didn't)

[the remark this answers is not in the document]
Comment (tmbo) [reply rasa_sdk/executor.py]: ah perfect :+1: changed it on master - thanks :+1:

[the remark this answers is not in the document]
Comment (tmbo) [reply rasa_sdk/executor.py]: and it only took me three commits to get this right 🤣

**Pull request merged**